diff --git a/contrib/location_cck/location_cck.module b/contrib/location_cck/location_cck.module
index 613dcaa..7f65675 100644
--- a/contrib/location_cck/location_cck.module
+++ b/contrib/location_cck/location_cck.module
@@ -391,7 +391,7 @@ function location_cck_field_widget_info() {
  */
 function location_cck_field_widget_form(&$form, &$form_state, $field, $instance, $langcode, $items, $delta, $element) {
   if ($field['type'] == 'location') {
-    $settings = $field['settings']['location_settings'];
+    $settings = isset($field['settings']['location_settings']) ? $field['settings']['location_settings'] : array();
 
     // Load location data for existing locations.
     if ($form_state['rebuild'] && !empty($form_state['values'][$field['field_name']][$langcode][$delta])) {
