diff --git a/database/zipcodes.us.mysql b/database/zipcodes.us.mysql
index 8ac80be..79dfe48 100644
--- a/database/zipcodes.us.mysql
+++ b/database/zipcodes.us.mysql
@@ -3498,8 +3498,10 @@ INSERT INTO zipcodes (zip, city, state, latitude, longitude, timezone, dst, coun
 ('10048', 'New York', 'NY', '40.712086', '-74.012270', -5, 1, 'us'),
 ('10055', 'New York', 'NY', '40.780751', '-73.977182', -5, 1, 'us'),
 ('10060', 'New York', 'NY', '40.780751', '-73.977182', -5, 1, 'us'),
+('10065', 'New York', 'NY', '40.764253', '-73.962466', -5, 1, 'us'),
 ('10069', 'New York', 'NY', '40.775477', '-73.990500', -5, 1, 'us'),
 ('10072', 'New York', 'NY', '40.780751', '-73.977182', -5, 1, 'us'),
+('10075', 'New York', 'NY', '40.773550', '-73.955621', -5, 1, 'us'),
 ('10079', 'New York', 'NY', '40.780751', '-73.977182', -5, 1, 'us'),
 ('10080', 'New York', 'NY', '40.780751', '-73.977182', -5, 1, 'us'),
 ('10081', 'New York', 'NY', '40.780751', '-73.977182', -5, 1, 'us'),
@@ -43193,4 +43195,4 @@ INSERT INTO zipcodes (zip, city, state, latitude, longitude, timezone, dst, coun
 ('99927', 'Point Baker', 'AK', '56.337957', '-133.606890', -9, 1, 'us'),
 ('99928', 'Ward Cove', 'AK', '55.395359', '-131.675370', -9, 1, 'us'),
 ('99929', 'Wrangell', 'AK', '56.409507', '-132.338220', -9, 1, 'us'),
-('99950', 'Ketchikan', 'AK', '55.875767', '-131.466330', -9, 1, 'us');
\ No newline at end of file
+('99950', 'Ketchikan', 'AK', '55.875767', '-131.466330', -9, 1, 'us');
